ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/role/AuditFilterAuthorizer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.filter.role;
+
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.ericsson.bss.cassandra.ecaudit.auth.AuditAuthorizer;
+import com.ericsson.bss.cassandra.ecaudit.utils.AuthenticatedUserUtil;
+import org.apache.cassandra.auth.AuthenticatedUser;
+import org.apache.cassandra.auth.IAuthorizer;
+import org.apache.cassandra.auth.IResource;
+import org.apache.cassandra.auth.Permission;
+import org.apache.cassandra.config.DatabaseDescriptor;
+
+public class AuditFilterAuthorizer
+{
+    private IAuthorizer authorizer; // lazy initialization
```

Comment:
Should this be volatile?

Reply:
I'm not sure. Will copies on all other CPU caches be invalidated if one thread is entering a synchronized block where this variable is modified? Or is the only guarantee that any modifications will be flushed to memory before sync block is entered?

Reply:
Never mind.